### PR TITLE
added capability to specify ebs snapshot by tag key and value pair

### PR DESCRIPTION
**Pull request by nlloyd**
- snapshotTag: driver option to specify tag value
- snapshotTagName: driver option to specify tag key (defaults to "Name")

### CODE REVIEW

File: package/ebs/rancher-ebs [hunk 1]
```diff
@@ -35,7 +35,8 @@ wait_volume_transition() {
         print_error "Failed volume ${VOLUME_ID} transition, expected end state is: ${end_state}, got ${current_state}"
     fi
     if [ $retriess -ge 10 ]; then
-        exit 2
+        print_error "Failed volume ${VOLUME_ID} transition, expected end state is: ${end_state}, got ${current_state} after ${retries} retries"
```
Comment (StrongMonkey) [inline package/ebs/rancher-ebs]: print_error will exit a different code https://github.com/rancher/storage/blob/master/package/common/common.sh#L85-L89.

File: package/ebs/rancher-ebs [hunk 2]
```diff
@@ -273,9 +300,10 @@ do_detach() {
     fi
     wait_volume_transition "in-use" "available"
     if [ $? -eq 2 ]; then
+        print_error "Attempting to detach ebs volume-id ${VOLUME_ID} by force."
```
Comment (StrongMonkey) [inline package/ebs/rancher-ebs]: same above. Do not use print_error for logging purpose.

File: package/ebs/rancher-ebs (hunk 1 above)
Comment (nlloyd) [reply package/ebs/rancher-ebs]: Swapped first print_error with log_error and second print_error with log_warn (which seemed appropriate for the context).

**Comment by StrongMonkey** [comment]
Thanks @nlloyd! LGTM, seems like ci for this repo is broken, we will revisit that back when we fix it.

**Comment by deniseschannon** [comment]
@nlloyd Could you rebase and try to re-push your commit? Another PR has fixed CI, but I need this PR to be green before merging

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/14315

**Comment by StrongMonkey** [comment]
LGTM

**Pull request merged**